### DISCUSSION TITLE
options to retrieve selected layer name in the mayaUsdLayerEditor ?

### DISCUSSION THREAD

**Discussion by voidreamer** [Autodesk/maya-usd#3682] [Q&A]
Hi all, I'm looking for a way to retrieve the current selected layer name (identifier) in the mayaUsdLayerEditor, I didn't see a flag for it in the `cmds.mayaUsdLayerEditorWindow` command, also the same to change layer selection by passing a layer identifier.
thanks!

**Comment by jcoelho88** [comment]
Hi @voidreamer 

If I'm understanding the question correctly you are asking to get the stages current EditTarget.
The EditTarget is the layer you can see selected via a blue circle in the mayaUsdLayerEditor.

To get the selected layer and its identifier you could run something like this in a python tab on the Maya script editor.
Also to switch the layer selection you can use stage.SetEditTarget(), this works so long as you know the path to the layer you want to switch to.

```python
import mayaUsd
from pxr import Sdf

maya_stage_node_list = cmds.ls(type='mayaUsdProxyShape', long=True)
stage = mayaUsd.ufe.getStage(maya_stage_node_list[0])

# Get the current edit target
edit_target_layer = stage.GetEditTarget().GetLayer()
print(edit_target_layer.identifier)

# Switch to a different edit target
new_layer = Sdf.Layer.FindOrOpen('/PATH/TO/LAYER.usd')
stage.SetEditTarget(new_layer)
```

Hope that answers the question.

Cheers,

Jason Coelho

**Reply by voidreamer** [reply]
Hi Jason, thanks for your answer,
I was looking rather for the selected layer item from the tree view on the Usd Layer Editor, like in this case the `lyr3`
for now I made a hacking way to get it by querying the qt pointer.
some times the edit layer is not the selected layer.

![image](https://github.com/Autodesk/maya-usd/assets/9424247/bb0e3031-364f-474f-8ac3-dc1f1a807145)

**Comment by wallworm** [comment]
This conversation fell off my radar, but we did add support for this last year. See the [LayerEditorWindowCommand](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/commands/Readme.md#LayerEditorWindowCommand) section of the docs. See -getSelectedLayers and -setSelectedLayers .
